client/i18n/i18next.ts

```diff
@@ -10,34 +12,42 @@ import { makeServerUrl } from '../network/server-url'
 
 const isDev = __WEBPACK_ENV.NODE_ENV !== 'production'
 
-export const i18nextPromise = i18n
-  .use(HttpBackend)
-  .use(initReactI18next)
-  .init<HttpBackendOptions>({
-    backend: {
-      loadPath: makeServerUrl('/locales/{{lng}}/{{ns}}.json'),
-      addPath: makeServerUrl('/locales/add/{{lng}}/{{ns}}'),
-    },
-
-    saveMissing: isDev,
-    saveMissingTo: 'all', // Save the missing keys to all languages
-
-    supportedLngs: ALL_TRANSLATION_LANGUAGES,
-    fallbackLng: TranslationLanguage.English,
-
-    // These are basically the defaults, but just defining them explicitly if we ever decide to use
-    // namespaces.
-    ns: TranslationNamespace.Global,
-    defaultNS: TranslationNamespace.Global,
-    fallbackNS: false,
-
-    interpolation: {
-      escapeValue: false, // Not needed for react as it escapes by default
-    },
-
-    // Some HTML attributes (e.g. `title`, `label`) only accept `string | undefined` as valid
-    // values, so we configure our `t` function to not be able to return `null` values.
-    returnNull: false,
+export const i18nextPromise = i18n.use(HttpBackend).use(initReactI18next)
+
+if (LANGUAGE_SUPPORT) {
+  const languageDetector = new LanguageDetector(null, {
+    order: ['localStorage', 'navigator'],
+    caches: ['localStorage'],
   })
 
+  i18n.use(languageDetector)
+}
+
+i18n.init<HttpBackendOptions>({
+  backend: {
+    loadPath: makeServerUrl('/locales/{{lng}}/{{ns}}.json'),
+    addPath: makeServerUrl('/locales/add/{{lng}}/{{ns}}'),
+  },
+
+  saveMissing: isDev,
+  saveMissingTo: 'all', // Save the missing keys to all languages
+
+  supportedLngs: ALL_TRANSLATION_LANGUAGES,
```

Comment:
If you want the feature flag to work you need to make this list contain only English don't you?

Reply:
And also probably remove the caches and such? Because otherwise if we were to ship this it would be setting people's languages to English even if there's a better language for them

Reply:
>If you want the feature flag to work you need to make this list contain only English don't you?

Hmm I thought it was fine because the feature flag prevents users to change their language and it disables language detector (so all users will basically use our fallback language, i.e. English).

Reply:
>And also probably remove the caches and such? Because otherwise if we were to ship this it would be setting people's languages to English even if there's a better language for them

Just tested this and it seems not to do anything? What the hell i18next... I'll have to investigate this. But anyway, removing the cache and the localStorage from options still adds an entry for the selected language to local storage for me.

Reply:
Yeah, if the language detector is disabled and nothing gets saved then that's fine. I don't know that we particularly need a feature flag for this anyway as it seems to all be implemented in this commit *anyway* but :)

(This feature flag is not useful for turning individual languages off anyway, so it being a "wait for more translations" thing isn't really useful)

Reply:
My main idea behind the feature flag was to not have an UI that allows users to select a language and then still seeing the content in English :p

Also, I was thinking we could merge this PR as is, and I would create a new PR for sending the language to login/signup APIs and saving it to the DB (and another PR after that for displaying that language in the settings page and being able to change it).

Reply:
Yeah but for what purpose exactly? The point of feature flags is to be able to implement things in pieces and get them checked in, even though parts of them are broken/non-working. All of these pieces work on their own once checked in, so I'm not really sure why we're adding the complexity of maintaining 2 implementations for all these things

Reply:
The flag doesn't help with the problem you stated (selecting a language and still seeing content in English). That will continue happening ~forever, every time we add new strings, those strings will be visible as English if they have not yet been translated. Every time we add a new language its strings will all be in English until they get translated. The feature flag system cannot fix this issue, it is not designed for that sort of problem

Reply:
Yes, all fair points :d
I'll take the feature flag out.